test/testcafe/spec/IdentifyRecoveryWithCaptcha_spec.js

```diff
@@ -0,0 +1,94 @@
+import { RequestMock, RequestLogger, Selector } from 'testcafe';
+import IdentityPageObject from '../framework/page-objects/IdentityPageObject';
+import { checkConsoleMessages } from '../framework/shared';
+import xhrIdentifyRecoveryWithRecaptcha from '../../../playground/mocks/data/idp/idx/identify-recovery-with-recaptcha-v2';
+import xhrIdentifyRecoveryWithHcaptcha from '../../../playground/mocks/data/idp/idx/identify-recovery-with-hcaptcha';
+import xhrAuthenticatorVerifySelect from '../../../playground/mocks/data/idp/idx/authenticator-verification-select-authenticator';
+
+const identifyRecoveryWithReCaptchaMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentifyRecoveryWithRecaptcha)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrAuthenticatorVerifySelect);
+
+const identifyRecoveryWithHCaptchaMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentifyRecoveryWithHcaptcha)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrAuthenticatorVerifySelect);
+
+const identifyRequestLogger = RequestLogger(
+  /idx\/identify/,
+  {
+    logRequestBody: true,
+    stringifyRequestBody: true,
+  }
+);
+
+const reCaptchaRequestLogger = RequestLogger(
+  /\/recaptcha\/api2\/userverify/,
+  {
+    logRequestBody: true,
+    stringifyRequestBody: true,
+  }
+);
+
+fixture('Identify Recovery - reset flow with Captcha');
+
+async function setup(t) {
+  const identityPage = new IdentityPageObject(t);
+  await identityPage.navigateToPage();
+  await checkConsoleMessages({
+    controller: 'forgot-password',
+    formName: 'identify-recovery',
+  });
+
+  return identityPage;
+}
+
+test.requestHooks(identifyRequestLogger, reCaptchaRequestLogger, identifyRecoveryWithReCaptchaMock)('should be able to submit identifier with reCaptcha enabled', async t => {
+  const identityPage = await setup(t);
+  
+  await identityPage.fillIdentifierField('test.identifier');
+  
+  // Wait for the reCaptcha container to appear in the DOM and become visible.
+  await t.expect(Selector('#captcha-container').find('.grecaptcha-badge').exists).ok({timeout: 3000});
```

Comment:
Not sure we discussed, trying to remember. Why this can't be rewritten in a way to avoid timeout ?

Reply:
These are DOM elements that get added to the page by the Captcha libs and since this happens on the fly and after all the related scripts load, we can't always guarantee that it will always be there without some buffer. Also, this timeout doesn't necessarily mean that the test will always wait 3s, if it detects the element before then it continues. It just guarantees that we wait **at most** 3s to check the selector: https://testcafe.io/documentation/402827/guides/concepts/built-in-wait-mechanisms#wait-mechanism-for-selectors

Reply:
But shouldn't it fail the expect and not proceed if it can't find the DOM element in the first place.. ?

Reply:
transferred this conversation over to Slack